feat(webapp): modernize development tooling - replace webpack, craco, with vite

benpsnyder opened the pull request:
Least amount of work possible to get `webapp` running with Vite.

Did minimal changes from #399

benpsnyder committed:
feat(webapp): modernize development tooling - replace webpack, craco, with vite and add support for tailwindcss

abouolia reviewed (commented):
To reduce the new additions on this PR, I suggest it would better if we split the replacing class to className part to separate PR. I would do tomorrow if you're busy.  @benpsnyder

benpsnyder commented on packages/webapp/src/containers/Setup/SetupSubscription/SetupSubscription.tsx:
```diff
@@ -7,25 +6,27 @@ import { SubscriptionPlansSection } from './SubscriptionPlansSection';
 import withSubscriptionPlansActions from '../../Subscriptions/withSubscriptionPlansActions';
 import styles from './SetupSubscription.module.scss';
 
+// TODO: use typescript https://github.com/NuroDev/lemonsqueezy.ts/tree/main/src/modules/subscription#-subscription
```
@abouolia kindly make the lemonsqueezy implementation type-safe. I had to disable this to get Vite to run. I can't test this functionality so I don't want to try to restore the functionality

benpsnyder commented:
> To reduce the new additions on this PR, I suggest it would better if we split the replacing class to className part to separate PR. I would do tomorrow if you're busy. @benpsnyder

Yes, the bulk of the file changes is a simple change from `class=` to `className=` but without the change, the UI doesn't render right.

if you leave it as `class=`
![image](https://github.com/bigcapitalhq/bigcapital/assets/707567/dd8ac724-edcb-4c4f-8c4e-5f4d138b0211)

changed to `className=` which is proper ...
![image](https://github.com/bigcapitalhq/bigcapital/assets/707567/7ce51c6c-a9c6-4b40-805f-53121a651d48)

benpsnyder committed:
Merge remote-tracking branch 'upstream/develop' into feat/webpack-to-vite

benpsnyder committed:
fix(repo): bump to node v18 and use pnpm in github workflow

abouolia commented:
I have looked at the PR, adding tailwindcss is something we avoided to do, for several reasons but the main reason because  Blueprintjs is our primary library components and BP does not support tailwindcss styling, we wanna follow the same styling approach of Blueprintjs to add consistency in our project. in the beginning we followed the styled components but now the CSS Module is used.

but the PR has a lot can benefit of it, for instance replacing "class=" to "className" becuase as you know className is more for React and we'd like to replace it on the whole project.

benpsnyder committed:
revert(repo): remove tailwind configurations

benpsnyder committed:
Merge remote-tracking branch 'origin/develop' into feat/webpack-to-vite
